src/Features/CSharp/Portable/Snippets/CSharpIfSnippetProvider.cs

```diff
@@ -38,7 +31,49 @@ protected override void GetIfStatementConditionAndCursorPosition(SyntaxNode node
         {
             var ifStatement = (IfStatementSyntax)node;
             condition = ifStatement.Condition;
-            cursorPositionNode = ifStatement.Statement.SpanStart + 1;
+            var blockStatement = (BlockSyntax)ifStatement.Statement;
+
+            var triviaSpan = blockStatement.CloseBraceToken.LeadingTrivia.Span;
+
+            // Getting the midpoint location of the trivia to find where the cursor shoudl end up
+            cursorPositionNode = triviaSpan.Start + (triviaSpan.Length / 2) + 1;
+        }
+
+        private static string GetIndentation(Document document, SyntaxNode node, SyntaxFormattingOptions syntaxFormattingOptions, CancellationToken cancellationToken)
+        {
+            var parsedDocument = ParsedDocument.CreateSynchronously(document, cancellationToken);
+            var ifStatementSyntax = (IfStatementSyntax)node;
+            var openBraceLine = parsedDocument.Text.Lines.GetLineFromPosition(ifStatementSyntax.Statement.SpanStart).LineNumber;
+
+            var indentationOptions = new IndentationOptions(syntaxFormattingOptions);
+            var newLine = indentationOptions.FormattingOptions.NewLine;
+
+            var indentationService = parsedDocument.LanguageServices.GetRequiredService<IIndentationService>();
+            var indentation = indentationService.GetIndentation(parsedDocument, openBraceLine + 1, indentationOptions, cancellationToken);
+            var newIndentation = new IndentationResult(indentation.BasePosition, indentation.Offset + syntaxFormattingOptions.TabSize);
+            return newIndentation.GetIndentationString(parsedDocument.Text, syntaxFormattingOptions.UseTabs, syntaxFormattingOptions.TabSize) + newLine;
+        }
+
+        protected override async Task<Document> AddIndentationToDocumentAsync(Document document, int position, ISyntaxFacts syntaxFacts, CancellationToken cancellationToken)
+        {
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var nearestStatement = FindAddedSnippetSyntaxNode(root, position, syntaxFacts);
+
+            if (nearestStatement is null)
```

Comment:
Do we ever expect this to be null?

Reply:
Ideally it shouldn't, but if something goes very wrong when trying to find the snippet we return null

Reply:
IMO we should throw on that case, and have `RoslynDebug.AssertNotNull` to get NRT correct

Reply:
I agree